### PR TITLE
Add `mc!tb` to check and update tie-breaker settings

### PR DESCRIPTION
**Pull request by AlphaKretin**
## Description

Allows configuration of tie-breaker settings to alleviate Challonge's undesirable defaults.

## Checklist

- [x] I am following the [contributing guidelines](https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/.github/CONTRIBUTING.md).
- [x] I have updated any relevant [user guides](https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/guides).
- [x] I have updated any relevant [documentation](https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/docs).
- [x] I have updated the [changelog](https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/changelog.md), or this change is not user-facing.

### CODE REVIEW

**Comment by AlphaKretin** [comment]
todos, amongst other: better interface, specify MB default on creation

File: src/commands/tb.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	"game wins": "Game/Set Wins",
+	"game win percentage": "Game/Set Win %",
+	"points scored": "Points Scored",
+	"points difference": "Points Difference",
+	"match wins vs tied": "Wins vs Tied Participants",
+	"median buchholz": "Median-Buchholz system"
+};
+
+const command: CommandDefinition = {
+	name: "forcescore",
```
Comment (kevinlul) [inline src/commands/tb.ts:20]: Name change

File: src/commands/tb.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import { CommandDefinition } from "../Command";
+import { TournamentStatus } from "../database/interface";
+import { reply } from "../util/discord";
+import { getLogger } from "../util/logger";
+import { challongeTieBreaker } from "../website/interface";
+
+const logger = getLogger("command:forcescore");
```
Comment (kevinlul) [inline src/commands/tb.ts:7]: Name change

File: src/website/interface.ts [hunk 3]
```diff
@@ -31,6 +32,15 @@ export interface WebsitePlayer {
 	seed: number;
 }
 
+export type challongeTieBreaker =
```
Comment (kevinlul) [inline src/website/interface.ts:35]: Capitalise this type

File: src/website/interface.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { ChallongeIDConflictError, UserError } from "../util/errors";
 export interface WebsiteWrapper {
 	createTournament(name: string, desc: string, url: string, topCut?: boolean): Promise<WebsiteTournament>;
 	updateTournament(tournamentId: string, name: string, desc: string): Promise<void>;
+	updateTieBreakers(tournamentId: string, tbs: challongeTieBreaker[]): Promise<void>;
```
Comment (kevinlul) [inline src/website/interface.ts:7]: A stub has to be in the mock thing too, even if we're not writing tests for this.

File: src/website/challonge.ts [hunk 5]
```diff
@@ -35,6 +35,7 @@ interface ChallongeTournamentSettings {
 	start_at?: Date;
 	check_in_duration?: number;
 	grand_finals_modifier?: null | "single match" | "skip";
+	tie_breaks?: string[]; // undocumented, hoping exists
```
Comment (kevinlul) [inline src/website/challonge.ts:38]: It does exist here, though we did have to inspect element to find all the values from the UI. https://api.challonge.com/v1/documents/tournaments/show

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: src/website/challonge.ts (hunk 5 above)
Comment (AlphaKretin) [reply src/website/challonge.ts:38]: This documents that it's an element returned from `tournaments/show`, not necessarily a setting that can be submitted with `tournaments/update` (or create, for that matter).

Comment (kevinlul) [reply src/website/challonge.ts:38]: Good point, I only tried reading it with `curl` when coming up with this idea. Then we should test it against update with curl first.

Comment (AlphaKretin) [reply src/website/challonge.ts:38]: Testing: it does work, using the values we found through inspect element. You don't have to provide exactly 3, fewer has defaults fill the gaps and more has extras ignored.

File: src/commands/tb.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+				tournament: id,
+				command: "tb",
+				event: "attempt"
+			})
+		);
+		const tournament = await support.database.authenticateHost(
+			id,
+			msg.author.id,
+			msg.guildID,
+			TournamentStatus.IPR
```
Comment (kevinlul) [inline src/commands/tb.ts:38]:
```suggestion
			TournamentStatus.PENDING
```

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: src/website/interface.ts [hunk 7]
```diff
@@ -39,6 +49,7 @@ export interface WebsiteTournament {
 	url: string;
 	players: WebsitePlayer[];
 	rounds: number;
+	tieBreaks: ChallongeTieBreaker[];
```
Comment (kevinlul) [inline src/website/interface.ts:52]: This very much can be `null` before the tournament is started, even though an ordering is shown in the UI.

Comment (AlphaKretin) [reply src/website/interface.ts:52]: Uh oh. In that case it's the challonge default?

**Review by kevinlul — APPROVED** [review]
LGTM!

**Pull request merged**